api/app/src/external/commonwell/document/document-query.ts

```diff
@@ -191,29 +190,51 @@ async function dowloadAndUploadDocsToS3({
 
           const data = await promise;
 
-          return {
+          const docWithFile: DocumentWithFilename = {
             ...doc,
             content: {
               ...doc.content,
               location: data.Location,
             },
             fileName: data.Key,
           };
-        }
 
-        return undefined;
+          const FHIRDocRef = toFHIRDocRef(docWithFile, organization, patient);
+          await upsertDocumentToFHIRServer(FHIRDocRef);
+
+          return FHIRDocRef;
+        }
       } catch (error) {
         capture.error(error, {
           extra: {
             context: `s3.documentUpload`,
           },
         });
         throw error;
+      } finally {
+        // TODO: eventually we will have to update this to support multiple HIEs
+        try {
+          const newPatient = await getPatientOrFail({ id: patient.id, cxId: patient.cxId });
+
+          await updateDocQuery({
+            patient: newPatient,
+            status: "processing",
+            progress: {
+              completed: completedCount + 1,
+              total: documents.length,
+            },
+          });
```

Comment:
Why are we updating this on the `finally` and not within the `try`? This way we'll be marking a failed document as completed.

I understand we don't want to leave a 99/100 indefinitely if something fails, but also saying that its 100/100 when one failed seems misleading.

Maybe we miss some sort of status per doc? Or maybe we just decrease the total number of files by one when a file fails to process?

Reply:
The way i see it is a way for the customer to know how many docs have been processed which means it could download or error out but it was still processed. 

I think in the future when we want to expose errors to the customers we can also send with the completed status how many errors out if that makes sense